### ISSUE TITLE
Add visualization for earmuff cone

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1897]
### Is your feature request related to a problem? Please describe.

It is really difficult to know what sliders are doing what because he can't see what is being modified it would be nice if we could

### Describe the solution you'd like

Some form of visualization of the earmuff mode settings

### Describe alternatives you've considered

guessing what they do

### Requesters

@epicEaston187

**Comment by ultrawidegamer** [comment]
I agree, as for right now I also have no idea what the earmuff settings values truly do without a visual indicator. This could be resolved by showing an indicator for a few seconds after any relevant setting has been updated

**Comment by epicEaston197** [comment]
Maybe the more information tab along with showing descriptions could also show two 2D graphics of what the earmuff settings sliders are doing
sense the cone is 3D two perspectives are needed in order for the settings to be properly visualized since the cone is on your head there's no good way of visualizing that that's why I am proposing this

**Comment by shiftyscales** [comment]
@Frooxius had mentioned that he expected more iteration to be needed with this feature.

I think that both for daily use of this feature, as well as debugging it/iterating on it, it would be helpful to have an option to visualize the earmuff range.

In VRC the user can choose whether the visualization is always active, or only active while editing the settings for its shape/size/distances.

**Comment by epicEaston197** [comment]
I want to note here that the visualization used in the video for debugging is not suitable because the debug visuals on your face without proper depth perception it just looks like a circle

**Comment by shiftyscales** [comment]
That's understood, @epicEaston197. Those visuals were made solely for the purposes of debugging/visualizing the system's function, not to be a pretty/public-facing feature.

**Comment by Hayden-Fluff** [comment]
I found this while searching around, and I've made a proof of concept visualization method, sort of inspired by VRChat's earmuff visualization.
https://storage.googleapis.com/haydens-video-storage/_temp/2024-12-31%2008-05-50.mp4

I've also exported it, it's a "facet" so you should be able to spawn it in userspace once saved.
https://storage.googleapis.com/haydens-video-storage/_temp/EarMuff%20Visualization.resonitepackage
*As a package since I am unable to save to my cloud*

**Comment by epicEaston197** [comment]
> I found this while searching around, and I've made a proof of concept visualization method, sort of inspired by VRChat's earmuff visualization. https://storage.googleapis.com/haydens-video-storage/_temp/2024-12-31%2008-05-50.mp4
> 
> I've also exported it, it's a "facet" so you should be able to spawn it in userspace once saved. https://storage.googleapis.com/haydens-video-storage/_temp/EarMuff%20Visualization.resonitepackage _As a package since I am unable to save to my cloud_

That is really sick
Is it toggle?

**Comment by Hayden-Fluff** [comment]
It does have a toggle, however it's a very sloppily thrown together proof of concept and it is not at all accurate from my very limited testing.

As for directionality, I'm not entirely sure how to visualize it, as I'm not entirely sure what it even does, nor do I really care enough to go into too much depth to figure it out.

I already spent a lot longer than I'd like on this and it was mostly just trying to figure out math I didn't even end up using.

**Comment by lxw404** [comment]
As far as I understand, directionality shifts spherical falloff into a spherical sector falloff (https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1944).  It is basically like a lerp between the two shapes.

**Comment by epicEaston197** [comment]
> As far as I understand, directionality shifts spherical falloff into a spherical sector falloff ([#1944](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1944)). It is basically like a lerp between the two shapes.

No see #1917 For more information this is a different feature It's how the audio falls off not the directionality of it

**Comment by SlyTheFloof** [comment]
really love this as i don't use earmuff mode like at all cus it's hard to set it up without visuals
